packages/server-wallet/src/wallet/store.ts

```diff
@@ -341,9 +346,19 @@ export class Store {
         )
     );
 
+    maybeChannel =
+      maybeChannel || (await timer('get channel', async () => getChannel(channelId, tx)));
+
+    if (!maybeChannel) {
+      // TODO: Discuss where to put the code that bumps a channel nonce (is here ok?)
+      await Nonce.fromJson({
+        value: wireSignedState.channelNonce,
+        addresses: wireSignedState.participants.map(p => p.signingAddress),
+      }).use(tx);
```

Comment:
I think whether or not this succeeds depends on the order in which new states are put in the store, something that I'm not sure I thought about when writing `Nonce.use`. It might need some thought/work.